consensus/scheme/rolldpos/rolldpos.go

```diff
@@ -217,21 +212,169 @@ func (r *RollDPoS) Metrics() (scheme.ConsensusMetrics, error) {
 
 // NumPendingEvts returns the number of pending events
 func (r *RollDPoS) NumPendingEvts() int {
-	return r.cfsm.NumPendingEvents()
+	if r.ctx.cfg.ParticipateConsensus {
```

Comment:
if ParticipateConsensus = false, should r.cfsm.NumPendingEvents() actually be 0?